### PR TITLE
Add validation for access levels on BBj class methods

### LINKED ISSUE #198: Respect method access-levels
```
class public Test 

    method private doTest()
    methodend
classend

t! = new Test()
t!.d...
```
image

_Originally posted by @StephanWald in https://github.com/BBx-Kitchen/bbj-language-server/issues/74#issuecomment-2340177699_

### PR DESCRIPTION
**Pull request by Lotes**
Closes #198

In the middle I refactored a `TypeInferer` class, but in the end it was not needed. Maybe we can use it later?

### CODE REVIEW

**Review by dhuebner — COMMENTED** [review]
See some comments above

File: bbj-vscode/src/language/bbj-validator.ts [hunk 1]
```diff
@@ -77,8 +78,47 @@ export class BBjValidator {
...
+                        remainingClasses.push(...c.extends.map(x => x.ref).filter(x => x!== undefined).map(x => x as Class))
+                    }
+                }
+            }
+        }
+        const member = memberCall.member.ref;
+        if(member && isBBjClassMember(member)) {
+            const actualAccessLevel = (member.visibility ?? "PUBLIC").toUpperCase();
+            if(!expectedAccessLevels.includes(actualAccessLevel)) {
+                accept('error', `The member '${member.name}' from the type '${classOfDeclaration.name}' is not visible`, {
```
[the remark this answers is not in the document]
Comment (Lotes) [reply bbj-vscode/src/language/bbj-validator.ts:120]: The code of method and classes are too different. The addressed duplicated code was dead, I forgot to erase it while moving it to the second file.
Message texts were adapted.

File: bbj-vscode/test/validation.test.ts [hunk 2]
```diff
@@ -211,4 +211,46 @@ describe('BBj validation', async () => {
             node: keywordStatement
         });
     });
+
+    test('Call methods with different access levels', async () => {
```
[the remark this answers is not in the document]
Comment (Lotes) [reply bbj-vscode/test/validation.test.ts]: > The CLASS verb may not be nested within any other CLASS, METHOD, or INTERFACE.

See [here](https://documentation.basis.cloud/BASISHelp/WebHelp/commands/class_verb.htm).

I added field and static cases.

**Review by dhuebner — APPROVED** [review]
LGTM, thanks!

**Pull request merged**